cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ServiceHttpServer.java

```diff
@@ -181,6 +206,22 @@ protected void shutDown() throws Exception {
     }
   }
 
+  public void setInstanceCount(int instanceCount) {
+    this.instanceCount = instanceCount;
+    this.contextFactory = createHttpServiceContextFactory();
```

Comment:
I think it's better to have the `instanceCount` as `AtomicInteger` and you just pass it to every `BasicHttpServiceContext` created by the factory. When the count change, simply update the value in the atomic integer and it will get reflected across all context objects ever created.